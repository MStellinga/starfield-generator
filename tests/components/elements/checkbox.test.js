/* eslint-disable jest/valid-expect */
import { h } from 'preact';
import Checkbox from '../../../src/components/elements/checkbox';
import render from 'preact-render-to-string';

describe('Test Checkbox', () => {
	test('Checkbox renders input with a label', () => {
        const tree = render(<Checkbox label="Test" />);
        expect(tree).toMatchSnapshot();	
    });
    
    test('Checkbox renders checked input with a label', () => {
        const tree = render(<Checkbox value="true" label="Test" />);
        expect(tree).toMatchSnapshot();	
	});
});
