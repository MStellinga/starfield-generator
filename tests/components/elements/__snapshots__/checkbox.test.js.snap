// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Checkbox Checkbox renders checked input with a label 1`] = `"<div class=\\"checkbox\\"><input type=\\"checkbox\\" checked=\\"true\\" /><span>Test</span></div>"`;

exports[`Test Checkbox Checkbox renders input with a label 1`] = `"<div class=\\"checkbox\\"><input type=\\"checkbox\\" /><span>Test</span></div>"`;
